src/main/scala/mesosphere/marathon/core/appinfo/impl/AppInfoBaseData.scala

```diff
@@ -122,11 +123,11 @@ class AppInfoBaseData(
 
     lazy val enrichedTasksFuture: Future[Seq[EnrichedTask]] = {
       def statusesToEnrichedTasks(
-        tasksById: Map[String, MarathonTask],
+        tasksById: Map[Task.Id, MarathonTask],
         statuses: Map[String, collection.Seq[Health]]): Seq[EnrichedTask] = {
         for {
           (taskId, healthResults) <- statuses.to[Seq]
-          task <- tasksById.get(taskId)
+          task <- tasksById.get(Task.Id(taskId))
```

Comment:
Generally my strategy towards the end of this work will be:

* Search for `MarathonTask`. Make sure it's only left as an implementation detail for storing `Task`s.
* Search for `Task.Id` conversions. Check whether we can get rid of them by converting some data structure from `String` to `Task.Id`.

To come up with incremental changes, I cannot do it all at once.